packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -894,6 +894,29 @@ const handleCfnFiles = (eventMap: EventMap, category: string, resource: string,
   });
 };
 
+const getNestedStackParameters = async (context: $TSContext, categories: Array<string>, amplifyMeta: any) => {
+  const s3 = await S3.getInstance(context);
+  const currentEnvName = context.amplify.getEnvInfo().envName;
+  const urlParametersMap = {};
+  for (const category of categories) {
+    const resources = Object.keys(amplifyMeta[category]);
+    for (const resource of resources) {
+      const resourceDetails = amplifyMeta[category][resource];
+      if (resourceDetails.providerMetadata) {
+        const templateURL = resourceDetails.providerMetadata.s3TemplateURL;
+        try {
+          const key = templateURL.split(resourceDetails.s3Bucket.deploymentBucketName + '/')[1];
+          const jsonBody = await s3.getFile({ Key: key }, currentEnvName);
+          urlParametersMap[templateURL] = Object.keys(JSON.parse(jsonBody.toString()).Parameters);
+        } catch {
+          urlParametersMap[templateURL] = [];
+        }
+      }
+    }
+  }
+  return urlParametersMap;
+};
```

Comment:
Couple of questions here.
- should we talk to S3 or read this from `#current-cloud-backend` directory?
- When we `amplify push --force` we ignore deployment bucket state deliberately. How this is going to work in force push context?